src/main/scala/com.snowplowanalytics/weather/providers/openweather/Errors.scala

```diff
@@ -19,39 +19,43 @@ object Errors {
   /**
    * Superclass for non-fatal exceptions that can be happen for weather fetching/processing
    */
-  sealed trait WeatherError {
-    val message: String
-    override def toString = s"OpenWeatherMap ${this.getClass.getSimpleName} " + message
+  sealed abstract class WeatherError(message: String) extends Throwable(message) {
+    override def toString = s"OpenWeatherMap ${this.getClass.getSimpleName}: $message"
```

Comment:
should we rather override `getMessage`?

Reply:
agreed

Reply:
I was too fast here, doing
``` scala
override def getMessage = s"OpenWeatherMap ${this.getClass.getSimpleName}: $message"
```
results following example error message when thrown:
``` scala
com.snowplowanalytics.weather.providers.openweather.Errors$ParseError: OpenWeatherMap ParseError: foo
```
I believe `toString` was overriden to remove long `com.snowplowanalytics.weather.providers.openweather.Errors` statement from output. Should we keep this behaviour or remove the overriding altogether (and just pass the message)?

Reply:
I think we're fine removing the overriding :+1:

Reply:
@chuwy do you remember its purpose?

Reply:
I'm almost sure it was done in order to have shorted message and I'm all for avoiding `toString` override, back then it didn't sound as a bad practice to me.